File: backend/schemas/conversation.py
from dataclasses import dataclass
from typing import Optional

from surrealdb import RecordID

from schemas.message import Message

@dataclass
class Conversation():
    id: Optional[RecordID]
    uuid: str
    title: str
    messages: list[Message]
    is_favorite: bool
    creation_timestamp: int
    last_message_timestamp: int

File: frontend/src/lib/schemas/conversation.ts
import type { RecordId } from "surrealdb"
import type { Message } from "./message"

export type Conversation = {
    id?: RecordId,
    uuid: string,
    title: string,
    is_favorite: boolean,
    creation_timestamp: number,
    last_message_timestamp: number,
    messages: Message[]
}
